### PR TITLE
"Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by HunterX-7**
## Dear Code reviewer In this Project, I:

- Followed [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Write meaningful commit messages in the most important parts of the process.
- Add the following features to the project:

  - Create a new Rails app called 'hello-rails-react'.
  - Use the command `rails new -j webpack my_new_rails_app` as described in the previous lesson's tutorial.
   - Verify that the `jsbundling-rails` gem is added to your Gemfile.
   - Initialize your project with Git.
   - Install needed node dependencies as described in the previous lesson's tutorial.
   - Create a `babel.config.js` in the root folder of your project as described in the previous lesson's tutorial.
   - Update `webpack.config.js` to include the babel loader as described in the previous lesson's tutorial.
   - Make sure that your project has a Postgres database set up. 
   - Create a table for storing your messages and create 5 different greetings. 
   - Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
   - Create a static view that will be the root of your app.
  - Create the `App` component with react-router and render it in your static view.
  - Create the `Greeting` component that will display a greeting. Set it up as a route in your App component.
  - Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
  - Display the random greeting in your `Greeting` component.

  Captura

## I Hope you have a great day and your support is much appreciated! 😄

### CODE REVIEW

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    dispatch(getGreeting());
+  }, [dispatch]);
+  
+  return (
+  <>
+    <h1 style={{textAlign: 'center'}}>Random Greeting Generator: </h1>
+    <h2 style={{textAlign: 'center', fontWeight: 'bold'}}>{greet}</h2>
+  </>
+  );
+};
```
Comment (anagudelogu) [inline app/javascript/components/App.js]: - [x] You did a great job working on this milestone 💯. However, according to the requirements:

image

Please make sure to use this `<App>` component to setup your routes, in this case, the one that is going to display the `<Greeting>` component. 🤓 🔧

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+})
+
+const greetingReducer = (state = [], action) => {
+    switch(action.type) {
+        case 'getGreeting/fulfilled':
+            return action.payload;
+        default:
+            return state;
+    }
+}
```
Comment (anagudelogu) [inline app/javascript/components/Greeting.js]: - [x] This logic shouldn't be inside the `components` folder as it represents state management with Redux. Please consider moving this to the `redux` folder. 🤓 🔧 

- [x] Also please consider moving the JSX you defined in your APP component to a separate `<Greeting>` component as per the requirements. 🤓 🔧

File: app/javascript/components/Greeting.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+export const getGreeting = createAsyncThunk('getGreeting', async() => {
+    const API_URL = 'http://localhost:3000/api/messages';
```
Comment (anagudelogu) [inline app/javascript/components/Greeting.js]: - [x] When I test both apps together this error comes up:

image

This is very common when you have client-server interaction, it basically means that your React client is not allowed to interact with your Rails server. Luckily it has a very easy fix, you can use the [`Rack-cors`](https://github.com/cyu/rack-cors) gem to allow your client to communicate with your server. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @HunterX-7, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/Greeting.js (hunk 3 above)
Comment (HunterX-7) [reply app/javascript/components/Greeting.js]: Thanks for your suggestions, the changes that you required have been made and everything seems to be running good with the rack cors gem. 😄 

Captura1

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @HunterX-7👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**